sunpy/instr/rhessi.py

```diff
@@ -219,7 +219,55 @@ def get_obssum_filename(time_range):
 
     filenames = dbase_dat.get('filename')[index_number_start:index_number_end]
     return [posixpath.join(get_base_url(), 'metadata', 'catalog', filename + 's')
-            for filename in filenames]
+        for filename in filenames]
+
+
+def get_obssum_filename_multiple_months(time_range):
+    """
+    Download the RHESSI observing summary data from one of the RHESSI
+    servers, parses it, and returns the name of the obssumm files relevant for
+    the time range.
+
+    Parameters
+    ----------
+    time_range : str, TimeRange
+        A TimeRange or time range compatible string
+
+    Returns
+    -------
+    out : list of lists
```

Comment:
why is this a list of lists? The examples shows only a list.

Reply:
Thats a typo in the example ,The function return list of lists with each list containing data for  a single month in the given time range . I'll just update it. Here is a code snippet on how this function works : 
```
>>> import sunpy.instr.rhessi as rhessi
>>> rhessi.get_obssum_filename_multiple_months(('2011/04/04', '2011/05/09'))
```
And the output was 
```
/home/yash_jain/Desktop/sunpy_env/sunpy/sunpy/instr/rhessi.py:100: UserWarning: Currently only support providing data from one whole day. Only data for 2011-05-01 will be returned
  _check_one_day(_time_range)
[['https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110405_031.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110406_041.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110407_031.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110408_038.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110409_030.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110410_028.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110411_024.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110412_024.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110413_022.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110414_017.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110415_020.fits', 'http://hessi.ssl.berkeley.edu/hessidata/metadata/catalog/hsi_obssumm_20110416_016.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110417_017.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110418_017.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110419_024.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110420_029.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110421_025.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110422_029.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110423_030.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110424_025.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110425_026.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110426_022.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110427_029.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110428_021.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110429_028.fits'], ['https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110501_021.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110502_023.fits', 'http://hessi.ssl.berkeley.edu/hessidata/metadata/catalog/hsi_obssumm_20110503_027.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110504_032.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110505_028.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110506_031.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110507_024.fits', 'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110508_022.fits']]
```
Filenames of different months in the time range as list

Reply:
I think it would be useful for the function to clean this up and make it just a simple list.